src/OpenTelemetry/Metrics/DataPoint/DataPoint.cs

```diff
@@ -16,65 +16,60 @@
 
 using System;
 using System.Collections.Generic;
-using System.Runtime.CompilerServices;
 
 namespace OpenTelemetry.Metrics
 {
     internal readonly struct DataPoint : IDataPoint
```

Comment:
It looks like `DataPoint` is currently only used by gauges at the moment. Is the plan to use this for other metric types as well?

Reply:
Also, would it make sense to call this `NumberDataPoint` aligning it with the naming from the data model?

Reply:
I'm open. However, I'm not 100% sure as we would need NumberDataPoint, SummaryDataPoint, HistogramDataPoint, etc. per each of our aggregator. But then, it ties our Aggregator to the OTLP protocol, which we may or may not want.

Reply:
Unless this is blocking, let's tackle this with an Issue/PR after this as necessary.